microsoft-edge/devtools-guide-chromium/memory-problems/memory-101.md

```diff
@@ -23,222 +23,175 @@ ms.date: 07/21/2023
 
-Learn [how to record a profile using the Heap Profiler](heap-snapshots.md).  In the following figure, some of the notable things in the Heap Snapshot recording in the [Memory](heap-snapshots.md) tool include **Distance**, which is the distance from the garbage-collection root.  If almost all the objects of the same type are at the same distance, and a few are at a bigger distance, that's something worth investigating.
+![Shallow and Retained size](./memory-101-images/shallow-retained.png)
 
-Distance from root:
+<!-- ------------------------------ -->
+#### Shallow size
 
-![Distance from root](./memory-101-images/root.png)
+The _shallow size_ is the size of the memory that is held by the object directly.
 
+The shallow size of an object is usually small because JavaScript objects often only store the descriptions of the object in memory, and not their values. Arrays and strings are exceptions to this and can have a significant shallow size. Most JavaScript objects store their values elsewhere in the renderer memory, and only expose a small wrapper object on the JavaScript heap.
 
-<!-- ====================================================================== -->
-## Dominators
+Nevertheless, even a small object can hold a large amount of memory indirectly, by preventing other objects from being disposed of by the garbage collection process.
 
-Dominator objects are comprised of a tree structure, because each object has exactly one dominator.  A dominator of an object might lack direct references to an object that it dominates.  That is, dominator's tree is not a spanning tree of the graph.
+<!-- ------------------------------ -->
+#### Retained size
 
-In the following figure:
+The _retained size_ is the size of the memory that's implicitly held by an object and that could be freed once the object and other existing retainers are deleted along with all the dependent objects that were made unreachable from GC roots.
 
-*  Node 1 dominates node 2.
-*  Node 2 dominates nodes 3, 4 and 6.
-*  Node 3 dominates node 5.
-*  Node 5 dominates node 8.
-*  Node 6 dominates node 7.
+That is, the retained size of an object is the amount of memory that would be regained if the object and all of its dependent objects were removed from the memory graph.
 
-![Dominator tree structure](./memory-101-images/dominatorsspanning.png)
+The retained size can't be smaller than the shallow size.
 
-In the following figure, node `#3` is the dominator of node `#10`.  But node `#7` also exists in every simple path from the garbage collection root **GC** to node `#10`. Therefore, an object `B` is a dominator of an object `A` if object `B` exists in every simple path from the root to the object `A`.
+When an object is retained by multiple nodes, the object's size appears in the retained size of the retainer node that has the shortest path to the GC root.
 
-Node `GC` dominates nodes `#1`, `#3`, and `#11`:
+<!-- ------------------------------ -->
+#### Distance
```

Comment:
re-order the 3 sections to match png at https://review.learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/memory-problems/memory-101?branch=pr-en-us-3033#object-sizes-and-distances 

1. Distance
2. Shallow size
3. Retained size